crates/server/src/message.rs

```diff
@@ -0,0 +1,55 @@
+use anyhow::Context;
+use lsp_types::notification::Notification;
+use std::sync::OnceLock;
+
+use crate::server::ClientSender;
+
+static MESSENGER: OnceLock<ClientSender> = OnceLock::new();
```

Comment:
In addition to `logging.rs`, this is the one other piece of global state we now have

This allows us to have free functions like `show_err_msg!()` that sends `ShowMessage` notifications to the client to pop up toast notifications for the user when something massive has gone wrong